cmd/api/src/api/v2/auth/auth_test.go

```diff
@@ -3137,3 +3168,187 @@ func TestActivateMFA_Success(t *testing.T) {
+			payload:              `{"token_name":"test token"}`, // user_id will be updated in the test
+			userContext: func(t *testing.T) (model.User, context.Context) {
+				admin, _ := createAdminUser(t)
+				// Using empty context to trigger error
+				return admin, context.Background()
+			},
+			setupMocks: func(t *testing.T, db *mocks.MockDatabase, req *http.Request) {
+			},
+		},
+		{
+			testName:             "Failed to store token",
+			expectedResponseBody: `{"http_status":500,"timestamp":"0001-01-01T00:00:00Z","request_id":"","errors":[{"context":"","message":"an internal error has occurred that is preventing the service from servicing this request"}]}`,
+			expectedResponseCode: http.StatusInternalServerError,
+			payload:              `{"token_name":"test token"}`, // user_id will be updated in the test
+			userContext: func(t *testing.T) (model.User, context.Context) {
+				admin, adminContext := createAdminUser(t)
+				return admin, adminContext
+			},
+			setupMocks: func(t *testing.T, db *mocks.MockDatabase, req *http.Request) {
+				user, _ := authz.GetUserFromAuthCtx(ctx.FromRequest(req).AuthCtx)
+				db.EXPECT().GetUser(gomock.Any(), user.ID).Return(user, nil)
+				db.EXPECT().CreateAuthToken(gomock.Any(), gomock.Any()).Return(model.AuthToken{}, errors.New("failed to store token"))
+			},
+		},
+		{
+			testName:             "Success",
+			expectedResponseCode: http.StatusOK,
+			payload:              `{"token_name":"test token"}`, // user_id will be updated in the test
+			userContext: func(t *testing.T) (model.User, context.Context) {
+				admin, adminContext := createAdminUser(t)
+				return admin, adminContext
+			},
+			setupMocks: func(t *testing.T, db *mocks.MockDatabase, req *http.Request) {
+				user, _ := authz.GetUserFromAuthCtx(ctx.FromRequest(req).AuthCtx)
+				db.EXPECT().GetUser(gomock.Any(), user.ID).Return(user, nil)
+
+				expectedToken := model.AuthToken{
+					UserID:     uuid.NullUUID{UUID: user.ID, Valid: true},
+					Name:       null.StringFrom("test token"),
+					HmacMethod: "hmac-sha2-256",
+					Key:        "test-key",
+				}
+
+				db.EXPECT().CreateAuthToken(gomock.Any(), gomock.Any()).Return(expectedToken, nil)
+			},
+		},
+	}
+
+	const endpointUrl = "/api/v2/tokens"
+
+	for _, testCase := range testData {
+		t.Run(testCase.testName, func(t *testing.T) {
+			// Setup
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+
+			resources, mockDB := apitest.NewAuthManagementResource(mockCtrl)
+			router := createRouter(endpointUrl, resources.CreateAuthToken, "POST")
+
+			user, ctx := testCase.userContext(t)
+
+			// For cases where we need to update the user_id in the payload
+			var payload string
+			if testCase.testName == "Success" || testCase.testName == "Failed to store token" {
+				var payloadMap map[string]interface{}
+				err := json.Unmarshal([]byte(testCase.payload), &payloadMap)
+				require.NoError(t, err)
+
+				payloadMap["user_id"] = user.ID.String()
```

Comment:
I was wrong in my first assessment of the payload comment. It looks like you did need to dynamically create the payload, in which case using a map makes a lot of sense. When I first read it it looked like a constant.